kernel/Kconfig

```diff
@@ -627,6 +627,12 @@ config SYS_CLOCK_EXISTS
 	  this is disabled.  Obviously timeout-related APIs will not
 	  work.
 
+config CYCLE_64BIT
```

Comment:
Based on the usage of this Kconfig symbol we should name with `HAS`. Probably we can use something like `CPU_HAS_CYCLE_64BIT`

Reply:
> `CPU_HAS_CYCLE_64BIT`

Not sure about the `CPU_` part, because it does not necessarily have to do with the **CPU** (especially for most of the system timer drivers that use a separate timer peripheral that is independent of the CPU).

Maybe `TIMER_HAS_CYCLE_64BIT` in `drivers/timer/Kconfig` that is selected by each timer driver (similar to `TICKLESS_CAPABLE`), and a separate config that enables 64-bit cycle support in the kernel (e.g. `CYCLE_64BIT`) that depends on `TIMER_HAS_CYCLE_64BIT`.